### DISCUSSION TITLE
Updates discussion

### DISCUSSION THREAD

**Discussion by autoSteve** [autoSteve/acMqtt#29] [Announcements]
Hi y'all.

autoSteve here.

I don't do the whole 'merge commit' thing on github. Rather I just direct commit to master, and this means that no notifications are sent to watchers when this happens. Ugh.

So I'm starting an 'announcements' discussion where I can spruik the latest and greatest changes (and bug fixes) to acMqtt.

Incremental changes seem to happen all the time, whether it's supporting a new application, like Unit Parameter, optimising to reduce load or improve performance, or just fixing something that's been broken for ages, like fan_pct was.

So to my announcement today, where reconnections are just better when a Mosquitto broker goes away and then comes back. While disconnected from the broker, inbound CBus messages now continue to be queued. When the broker connects again the queued messages are delivered in order. Because these messages are no longer lost, a full and unnecessary re-publish of discovery and current state topics is no longer performed after a successful reconnection.

And I think that is a cool improvement.

Cheers.

@lee-booy, @notsomot, @arrikhan, @vworp, @sgraystar, @pbyrne-ms, @ziogref, @Nick-Adams-AU, @gitcoinhello1, @hmousaab, @Hosemans, @srpage, @Warren2006, @sthlmsrille, @mjohnson9, @rjmidau, @nicpullan, @atomic10, @TheSeaCaptn

**Comment by autoSteve** [comment]
Thanks to @arrikhan pointing out a flaw with 'MQTT send receive' cover status in Home Assistant, covers have had a makeover.

There are now three alternative modes of operation, including using a basic slider for level, a level translation mode up/down/stop with rather crude HA status, and now a third alternative, which is level translation with status tracking.

Check out the readme for the full details, but in short with a bit of simple calibration, a rate can now be set that allows 'MQTT send receive' to update HA in near real time with a quite accurate approximation of where a cover is on its journey up or down.

Cheers,
Steve

**Comment by autoSteve** [comment]
A blow for simplicity.

Event-based scripts are all gone, leaving just residents.

This simplifies initial deplyment, plus the number of scripts to configure/update. Using the localbus instead of socket library seems to have lowered Automation Controller load as a bonus.

I'm also switching to a create branch/merge discipline on GitHub, so that notifications will get sent to watchers on update. So if you've not yet watched acMqtt then now might be a good time to do so.

Steve

**Reply by lee-booy** [reply]
Very impressive and great to see it evolving.
- Are there any downsides to this approach?
- I assume there is no impact to the devices in HA with migrating to the new single script?

**Reply by autoSteve** [reply]
I've not encountered any downside, @lee-booy, and function should be identical so business as usual for Home Assistant.

So far, in my environment the only "impact" that I've seen is a positive one, with Automation Controller load dropping. On the back of that I'm migrating all of my other socket-dependent resident/event pairs to use the localbus.

It took a bit of investigation to work out how to replicate the no "Execute during ramping" behaviour that event scripts feature, which was a bit more involved than ticking a box. But that now works just fine. I've not yet tested that on firmware prior to 1.15.0, where execute during ramp for event scripts was broken (it always did, regardless of the tick box).

<img width="593" alt="image" src="https://github.com/autoSteve/acMqtt/assets/37229860/616e6106-1c1b-4b38-8785-6f1eb4e8b089">

You should be able to put the new code in a separate resident, disable current, enable new testing code. The new code will disable any event script (MQTT/MQTT final) on startup, so it won't intefere. To revert, disable new and enable old, which on startup will reenable the event script.

**Reply by ziogref** [reply]
I'm running version 1.14 and was able to update the scripts. 

I just had to set this to false `local entityIdAsIndentifier` as all my stuff was using the `light.cbus_mqtt_254_56_10` naming convention. My cbus stuff is now all working within Home Assistant.

**Reply by autoSteve** [reply]
I should have created an updates discussion topic about that one, as it's quite a significant change . Oops. If the IDs are changed to the new style then dashboards that reference them will break.

As @ziogref found, @lee-booy, just change this variable at the top of the script, restart, and all dashboards should be back to normal.

**Reply by autoSteve** [reply]
@ziogref, I'd be interested to hear how C-Bus ramps get represented in HA when using 1.14.0. e.g. Start a reasonably long ramp on the AC, and observe status in HA. At the end of the ramp the status should change in HA, and not before. There should be no updates during the ramp in HA.

**Reply by lee-booy** [reply]
I'm already on 1.15 and using the new entityIdAsIdentifier so should be no impact for me 🚀

**Reply by autoSteve** [reply]
That's right. You requested it! 🤦🏻🤣

**Reply by ziogref** [reply]
@autoSteve CBUS only controls my lights now, nothing else.

**Reply by ziogref** [reply]
@autoSteve Set one of my lights to turn on over 30s. HA didn't report the light was on until it was 100% (255) on.

**Comment by autoSteve** [comment]
_MQTT send receive_ now features a way to have entities named as their identifier. For example light.kitchen_main_light, instead of light.cbus_mqtt_254_56_1.

I think this is a terrific change (thank you @lee-booy), but it will break dashboards that refer to the old entity names if upgrading the script, requiring work to fix them.

If you hate this change, or just want to put off the required work, the variable `local entityIdAsIndentifier` near the top of the script controls this behaviour. Set it to `false` to use the old entity names and restart, and things should be good.

**Comment by autoSteve** [comment]
Inbound sensors are now added to _MQTT send receive_ (thank you Michael Brooks). An example could be the state of charge of a Tesla Powerwall in Home Assistant being pushed to the automation controller as a user parameter, which could then be monitored to shed C-Bus loads. Or the user parameter could just be displayed for information on a visualisation home page.

Set up an automation in HA to publish sensors to MQTT on state change, as noted in an example in README.md.

Use keywords like `MQTT, isensor, topic=powerwall/soc`

**Comment by autoSteve** [comment]
Insidious bug fix time. Thanks @sgraystar.

The localbus callback in _MQTT send receive_ now correctly converts datahex to almost all values, getting the correct value for all localbus events. In the prior code, grp.getvalue() was used for non-lighting app, and on my NAC (relatively slow compared to @sgraystar's shiny new AC2) would almost always return the correct value. Given a faster processor this was almost never the case.

So a heartfelt thanks to @sgraystar for having the patience to work through getting C-Bus event things (hopefully) absolutely right. Good on ya mate.

(I gotta get me one of these new generation ACs...)

**Comment by autoSteve** [comment]
FYI, new Automation Controller firmware v1.17.0 was released by Schneider in May, containing a seemingly essential bug fix. 1.15.0/1.16.0 have been removed from downloads.

**Comment by autoSteve** [comment]
With thanks to @pbyrne-ms, I give you `state_class=` and generic discovery variables via `disco=`.

Nice one, Paddy.

**Comment by autoSteve** [comment]
Script updated to allow a device class to be set for `binary_sensor` using `class=`. This allows binaries states other than on/off, and automatic selection of a dynamic icon.

**Reply by jasonhollis** [reply]
Steve,

Out of curiosity do you have the use case or scenario that is the driver behind the update sir?

**Reply by autoSteve** [reply]
Read from here, @jasonhollis. https://github.com/autoSteve/acMqtt/discussions/60#discussioncomment-12906041

**Comment by lzj719981506** [comment]
Hello, I have an old 5500NAC, and now I need its firmware version, which is less than 1.10.0. If there is, please email me: 18820059613@139.com. Thank you🙏
